### PR TITLE
[ONNX export] Fixing spatial export for batchnorm

### PR DESCRIPTION
**Pull request by vinitra**
## Description ##
In the ONNX model zoo, we noticed that models like ArcFace and DUC that have been exported from mxnet with batchnorm operators are not treating spatial mode correctly. Without this PR, this model export pipeline is broken.

https://github.com/onnx/models/issues/156
https://github.com/onnx/models/issues/91#issuecomment-540312125

cc: @ankkhedia @hariharans29 @snnn

Quoting from [MxNet BatchNorm documentation](https://mxnet.apache.org/api/python/docs/tutorials/packages/gluon/training/normalization/index.html): "One of the most popular normalization techniques is Batch Normalization, usually called BatchNorm for short. We normalize the activations across all samples in a batch for each of the channels independently."

The comment in the exporter refers to mean and variance per feature, instead of per channel. Fixing this will mean that spatial mode should be 1, instead of 0 in the ONNX export.

Changing spatial to 1 fixed these models, in accordance with the issue referenced above.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Fixing onnx export for batchnorm spatial mode

### CODE REVIEW

**Comment by snnn** [comment]
LGTM.

**Comment by vinitra** [comment]
@szha, can you take a look?

**Comment by RuRo** [comment]
As far as I understand, `spatial=1` is the default in all versions of the `BatchNormalization` operator. The newer versions don't have this attribute, but still do a computation as if `spatial=1`, so I think we are better off removing the explicit attribute and relying on the default behaviour, since specifying it explicitly will make it harder to convert to newer ONNX opset versions.

What do you think?

**Comment by vinitra** [comment]
> As far as I understand, `spatial=1` is the default in all versions of the `BatchNormalization` operator. The newer versions don't have this attribute, but still do a computation as if `spatial=1`, so I think we are better off removing the explicit attribute and relying on the default behaviour, since specifying it explicitly will make it harder to convert to newer ONNX opset versions.
> 
> What do you think?
> 
> Edit:
> 
> The comment says that the default for ONNX is to compute `across all spatial features`, but the docs [here](https://github.com/onnx/onnx/blob/master/docs/Changelog.md) say `spatial : int (default is 1)` for all BatchNormalization ops that have this attribute. Admittedly, the ONNX documentation wording is really confusing.

This makes sense! I've deprecated the attribute in my recent PR update, and provided more clarity in the comments.

**Comment by vinitra** [comment]
@haojin2 @hzfan -- this PR has been waiting for a review for a while! Can you take a look / refer us to the right folks? I noticed you have been active approvers of recent PRs.

**Comment by sandeep-krishnamurthy** [comment]
@vandanavk - Can you please help review this changes? Thanks!

**Comment by sandeep-krishnamurthy** [comment]
@vinitra - Thanks for your contributions. can you please sync to master?

**Comment by vinitra** [comment]
> @vinitra - Thanks for your contributions. can you please sync to master?

Just updated! Thanks for the review.

**Comment by sandeep-krishnamurthy** [comment]
@mxnet-bot run ci [unix-gpu, windows-gpu]

**Comment by sandeep-krishnamurthy** [comment]
Jenkins CI successfully triggered : [windows-gpu]

**Comment by sandeep-krishnamurthy** [comment]
@ChaiBapchya - windows-gpu build is failing consistently here. Any suggestions please? Thanks.

**Comment by snnn** [comment]
I don't know why it just started failing, but I know how to fix it.

The root cause is: you're using 32 bits compiler for 64 bits target.  A 32 bits program can only have 2GB address space, so you may run out it quickly.
```
[2020-03-25T21:40:57.615Z] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
```

Please add "-A x64 -T host=x64" to your cmake command line args.

**Comment by ChaiBapchya** [comment]
@snnn  Thanks for the suggestion. Actually @leezu in his https://github.com/apache/incubator-mxnet/pull/17912/ tried that toolchain fix but that still failed (I think coz of removing JOM)..

**Comment by vinitra** [comment]
@mxnet-bot run ci [unix-gpu, windows-gpu]

**Comment by vinitra** [comment]
@mxnet-bot run ci [all]

**Comment by vinitra** [comment]
Waiting on https://github.com/apache/incubator-mxnet/pull/17962 for CI fixes.

**Comment by sandeep-krishnamurthy** [comment]
@vinitra - Sorry for the inconvenience caused. A fix for windows-gpu is now submitted.
Can I please request to rebase this PR so the change is reflected here?
Thanks again for the contribution and apologies as your PR took too long to be merged.

**Comment by vinitra** [comment]
> @vinitra - Sorry for the inconvenience caused. A fix for windows-gpu is now submitted.
> Can I please request to rebase this PR so the change is reflected here?
> Thanks again for the contribution and apologies as your PR took too long to be merged.

No worries! Thanks for the help.

**Comment by sandeep-krishnamurthy** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by vinitra** [comment]
@mxnet-bot run ci [macosx-x86_64]

**Comment by ChaiBapchya** [comment]
@vinitra haven't configured mxnet-bot to trigger the macosx yet. [It's not mandatory for merge either]. But if needed we can increase the scope to incorporate that trigger.

**Comment by sandeep-krishnamurthy** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by ChaiBapchya** [comment]
@sandeep-krishnamurthy This fix got merged in master. I checked it's neither in v1.7.x nor in previous releases [e.g. 1.6.0]
So customers trying to export MX models [which include batchnorm op] to ONNX would fail for all release binaries[such as 1.5.1, 1.6.0, 1.7.0] 
If we cherry-pick this into the 1.x & v1.7.x branch [atleast nightly binaries won't have this issue]
